meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -135,7 +130,7 @@ if test "${fiovb.is_secondary_boot}" = "0"; then
 			run rollback_setup
 			run saveenv_mmc
 		else
-			if test "${fiovb.bootupgrade_available}" = "1"; then
+			if test ${fiovb.bootupgrade_available} -gt 0; then
```

Comment:
We need to have "finalized" the boot firmware OTA prior to hitting this point (moving the secondary path into the primary path and performing a reset).  If we don't then we overwrite the secondary path losing it's original values entirely.  What is current in the primary path is actually firmware from N-2 OTA.

Reply:
@mike-scott we don't need finalization, as we don't have `bootupgrade_primary_updated` for regular boot-common.
Even if OTA sysroot will be updated during secondary boot path, after reboot from Linux we will boot primary path, put latest boot firmware from latest OTA sysroot, and reboot again into secondary path to validate it.